### PR TITLE
Rotate Kernel

### PR DESCRIPTION
**Pull request by christopherngutierrez**
## Proposed changes

Added rotate kernel, which has been tested and validated to work.

Minor changes (to avoid code replication):
- Created a helper function for common Polys in Relin/Rotate kernels
- Added a fix index to the KeyMul, which is different for Relin/Rotate kernels

## Types of changes

What types of changes does your code introduce to the HE Toolkit project?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you are unsure about any of them, do not hesitate to ask. We are
here to help! This is simply a reminder of what we are going to look for before
merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/IntelLabs/hec-p-isa-tools/blob/main/CONTRIBUTING.md) agreement
- [ ] Current formatting and unit tests / base functionality passes locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by kylanerace — APPROVED** [review]
LGTM, tested and running E2E

**Comment by faberga** [comment]
This PR is dependent on PR #43 and should not be merged before it.

File: kerngen/pisa_generators/relin.py [hunk 1]
```diff
@@ -120,28 +138,24 @@ def to_pisa(self) -> list[PIsaOp]:
...
-        upto_last_coeffs.start_parts = 0
+        input_last_part, last_coeff, upto_last_coeffs = init_common_polys(
+            self.input0, self.context.key_rns
+        )
 
         add_original = Polys.from_polys(mul_by_rlk_modded_down)
         add_original.name = self.input0.name
 
-        return mixed_to_pisa_ops(
+        opts = mixed_to_pisa_ops(
```
Comment (faberga) [inline kerngen/pisa_generators/relin.py:149]: This was resolved in the Relin PR but resurface in the Rotate. Sounds as the rotate code has not been rebased correctly.

File: kerngen/pisa_generators/relin.py [hunk 2]
```diff
@@ -120,28 +138,24 @@ def to_pisa(self) -> list[PIsaOp]:
...
             Comment("Multiply by relin key"),
-            KeyMul(self.context, mul_by_rlk, upto_last_coeffs, relin_key),
+            KeyMul(self.context, mul_by_rlk, upto_last_coeffs, relin_key, 2),
             Comment("Mod switch down to Q"),
             Mod(self.context, mul_by_rlk_modded_down, mul_by_rlk),
             Comment("Add to original poly"),
             Add(self.context, self.output, mul_by_rlk_modded_down, add_original),
             Comment("End of relin kernel"),
         )
+        return opts
```
Comment (faberga) [inline kerngen/pisa_generators/relin.py:161]: Same as above.

**Review by faberga — CHANGES_REQUESTED** [review]

**Comment by faberga** [comment]
Closed as this PR has been superseded by PR #48

**Pull request closed without merging**